src/commands.ts

```diff
@@ -91,6 +92,139 @@ export async function updateDependencies(ctx: WorkspaceContext) {
     await updateFolderDependencies(current);
 }
 
+/**
+ * Prompts the user to input project details and then executes `swift package init`
+ * to create the project.
+ */
+export async function createProject(ctx: WorkspaceContext): Promise<void> {
+    // Prompt the user for the type of project they would like to create
+    const availableProjectTemplates = await ctx.toolchain.getProjectTemplates();
+    const selectedProjectTemplate = await vscode.window.showQuickPick<
+        vscode.QuickPickItem & { type: SwiftProjectTemplate }
+    >(
+        availableProjectTemplates.map(type => ({
+            label: type.name,
+            description: type.id,
+            detail: type.description,
+            type,
+        })),
+        {
+            placeHolder: "Select a swift project template",
+        }
+    );
+    if (!selectedProjectTemplate) {
+        return undefined;
+    }
+    const projectType = selectedProjectTemplate.type.id;
+
+    // Prompt the user for a location in which to create the new project
+    const selectedFolder = await vscode.window.showOpenDialog({
+        title: "Select a folder to create a new swift project in",
+        openLabel: "Select folder",
+        canSelectFiles: false,
+        canSelectFolders: true,
+        canSelectMany: false,
+    });
+    if (!selectedFolder || selectedFolder.length === 0) {
+        return undefined;
+    }
+
+    // Prompt the user for the project name
+    const existingNames = await fs.readdir(selectedFolder[0].fsPath, { encoding: "utf-8" });
+    let initialValue = `swift-${projectType}`;
+    for (let i = 1; ; i++) {
+        if (!existingNames.includes(initialValue)) {
+            break;
+        }
+        initialValue = `swift-${projectType}-${i}`;
+    }
+    const projectName = await vscode.window.showInputBox({
+        value: initialValue,
+        prompt: "Enter a name for your new swift project",
+        validateInput(value) {
+            // Ensure there are no name collisions
+            if (existingNames.includes(value)) {
+                return "A file/folder with this name already exists";
+            }
+            return undefined;
+        },
+    });
+    if (!projectName) {
+        return undefined;
+    }
+
+    // Create the folder that will store the new project
+    const projectUri = vscode.Uri.joinPath(selectedFolder[0], projectName);
+    await fs.mkdir(projectUri.fsPath);
+
+    // Use swift package manager to initialize the swift project
+    await withDelayedProgress(
+        {
```

Comment:
Given there is no progress reported here, any reason you can't use `WorkspaceContext.statusItem.showStatusWhileRunning()`

Reply:
I'd expect that 99% of the time we would never even show the notification. In my testing it never showed up unless I reduced the timeout. However, if it does take longer than expected, I want it to be a prominent notification rather than something hidden away in the status bar. That way users know that something is actually happening in the background.

That being said, this does add extra code that will need to be maintained. I'm not too strongly attached to this opinion and can just use showStatusWhileRunning() instead if that makes more sense to you.